src/runtime/formal/ops/ops.cc

```diff
@@ -562,37 +530,32 @@ CVM_REGISTER_GLOBAL("cvm.runtime.formal.strided_slice")
 
     for (size_t i = 0; i < begin_vec.size(); ++i) {
       int64_t begin_range = stride_vec[i] < 0 ? -1 : 0;
-      int64_t end_range = stride_vec[i] < 0 ? dshp[i] -1 : dshp[i];
+      int64_t end_range = stride_vec[i] < 0 ? xShape[i] -1 : xShape[i];
       int64_t begin = begin_vec[i];
-      if (begin < 0) begin += dshp[i];
+      if (begin < 0) begin += xShape[i];
       begin_vec[i]= std::min(std::max(begin, begin_range), end_range);
     }
 
-//<<<<<<< HEAD
-//    TShape xShape(x->shape, x->shape + x->ndim),
-//        yShape(y->shape, y->shape + y->ndim);
-//    for(uint64_t i = 0; i < getSize(y); i++){
-//      TShape yVecIndex = VectorIndex(yShape, i);
-//      TShape xVecIndex(ndim);
-//      for (int j = 0; j < xVecIndex.ndim(); j++) {
-//        xVecIndex[j] = begin_vec[j] + stride_vec[j] * yVecIndex[j];
-//      }
-//      y_data[i] = x_data[ScalarIndex(xShape, xVecIndex)];
-//=======
-    for(uint64_t i = 0; i < CVMArgSize(args[1]); i++){
-        uint64_t o_i = i, in_i = 0, shapeSize = 1;
-        for(int j = ndim-1; j >= 0; j--){
-            uint64_t col = o_i % y->shape[j];
-            o_i /= y->shape[j];
-            int64_t tbegin = begin_vec[j];
-            int64_t tstep = stride_vec[j];
-            col = tbegin + col * tstep;
-            in_i += col * shapeSize;
-            shapeSize *= x->shape[j];
-        }
-        y_data[i] = x_data[in_i];
-//>>>>>>> wlt
+    Indices xIdx(xShape);
+    for (Indices yIdx(yShape); !yIdx.End(); yIdx++) {
+      for (int i = 0; i < xIdx.ndim(); i++) {
+        xIdx[i] = begin_vec[i] + stride_vec[i] * yIdx[i];
```

Comment:
The index of `yIdx` will cause recalculate the index cache invalid, but it's not necessary.

If you really have the requirement of accessing for value in `yIdx`, you can invoke other function
to get a copy instead of a reference. 

Maybe you'd like to define a new member function like `At` or cast the non-const object to const
before indexing via `static_cast<const Indices&>(yIdx)[i]` to get value.

Reply:
Remove the non-const operator[] function and reserve the const return version.

Add an extra function `Ref()` to get the reference value.